apps/web/app/courses/[courseCode]/page.tsx

```diff
@@ -0,0 +1,185 @@
+"use client";
+
+import { Center, Container, Divider, Flex, Pagination, Stack, Title, Text, Loader, Alert } from "@mantine/core";
+import { MyReviewCard, ReviewCard } from "@components/ui/review-card";
+import { useEffect, useState } from "react";
+import { PaginatedResponse } from "types/pagination";
+import { Course } from "types/course";
+import { Review } from "types/reviews";
+import { useDisclosure } from "@mantine/hooks";
+import Link from "next/link";
+import { NotificationData, notifications } from "@mantine/notifications";
+import WriteReviewForm from "@components/ui/write-review-form";
+import SessionModal from "@components/ui/session-modal";
+import { Session } from "@supabase/supabase-js";
+import CourseComposeAPIClient from "lib/api/api";
+import { useAuth } from "hooks/use-auth";
+import { IconLock } from "@tabler/icons-react";
+
+export default function CourseReview({ params }: { params: { courseCode: string } }) {
+	const [courseData, setCourseData] = useState<Course>();
+	const [reviewsData, setReviewsData] = useState<PaginatedResponse<Review>>();
+	const [sessionData, setSessionData] = useState<Session | null>();
+	const [showReviewLimitAlert, setShowReviewLimitAlert] = useState(false);
+	const [isLoading, setIsLoading] = useState(true);
+	const [pageNumber, setPageNumber] = useState(1);
+
+	const { getSession } = useAuth();
+
+	const [opened, { open: openSessionModal, close: closeSessionModal }] = useDisclosure(false);
+
+	const apiClient = new CourseComposeAPIClient(params.courseCode);
+
+	const handleSubmitResponse = (result: NotificationData) => {
+		if (!result.title || !result.message) {
+			// inform user to re-login when submit with missing or expired token
+			openSessionModal();
+			return;
+		}
+
+		notifications.show(result);
+		setTimeout(() => window.location.reload(), 3000);
+	};
+
+	useEffect(() => {
+		apiClient
+			.fetchCourseDetails()
+			.then((course) => setCourseData(course))
+			.catch(console.error); // TODO: handle if no course found? - maybe redirect to home page
+
+		setIsLoading(true);
+
+		getSession()
+			.then((session) => {
+				setSessionData(session);
+				if (!session) openSessionModal();
+			})
+			.catch(console.error)
+			.finally(() => {
+				setIsLoading(false);
+			});
+	}, []);
```

Comment:
```suggestion
	}, [apiClient, getSession, openSessionModal]);
```

Reply:
Tried adding either apiCilent and getSession and it's causing infinite fetching. Not sure why?

Reply:
Yeah, 'cuz the apiClient instance is created on each render.. it's instantiated inside the component body LOL